### PR TITLE
Dynamically create scrape DAGs from a config file

### PR DESCRIPTION
**Pull request by hancush**
## Description

This PR revises our approach to scheduling. Rather than making use of a branch operator, which tightly couples DAG outcomes with the day and time they're run, it moves all scheduling into the cron layer. Because Airflow doesn't natively allow multiple schedule intervals for the same DAG, we achieve this by creating a config object containing each type of scrape and its schedule, then iterating over it in `scrape_factory.py` to dynamically generate the corresponding DAGs. The only exceptions to this pattern are `full_scrape` (which I've refactored into a DAG with a separate task for each scrape, so scraped data is cleared out between each entity type, hopefully addressing #63 and #64) and the ETL DAGs, `hourly_processing` and `refresh_guid`.

Handles #59.

## Testing instructions

- Remove any Airflow volumes you already have: `docker-compose down --volumes`
- Run the Metro app and Airflow instance as described in the README.
- Navigate to the Airflow instance at http://localhost:8080 (logging in with the dummy credentials, `foo` / `bar`) and confirm you see a DAG for every scrape and schedule interval, plus the full scrape and ETL DAGs.

Screen Shot 2020-12-09 at 9 08 46 AM

N.b., I've triggered the smaller scrapes locally to confirm the revised pattern works. After review, I'd like to deploy this to staging and let it run for a couple of days to test that it's working as expected in a deployment environment.

### CODE REVIEW

File: dags/scrape_factory.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+docker_base_environment = {
+    'DECRYPTED_SETTINGS': 'pupa_settings.py',
+    'DESTINATION_SETTINGS': 'pupa_settings.py',
+    'DATABASE_URL': LA_METRO_DATABASE_URL,  # For use by entrypoint
+    'LA_METRO_DATABASE_URL': LA_METRO_DATABASE_URL,  # For use in scraping scripts
+}
+
+for dag_name, dag_config in SCRAPING_DAGS.items():
+    for idx, interval in enumerate(dag_config['schedule_interval'], start=1):
+        dag_id = dag_name if len(dag_config['schedule_interval']) == 1 else '{0}_{1}'.format(dag_name, idx)
```
Comment (hancush) [inline dags/scrape_factory.py:20]: This was a quick and dirty way to give DAGs unique names if we have multiple schedules. We might want to do something more description, e.g., parse the weekday from the cron expression.

**Comment by fgregg** [comment]
i think this is a worthy approach!

let me know if/when you want a code review.

File: dags/full_scrape.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+docker_base_environment = {
+    'DECRYPTED_SETTINGS': 'pupa_settings.py',
+    'DESTINATION_SETTINGS': 'pupa_settings.py',
+    'DATABASE_URL': LA_METRO_DATABASE_URL,  # For use by entrypoint
+    'LA_METRO_DATABASE_URL': LA_METRO_DATABASE_URL,  # For use in scraping scripts
+    'RPM': 60,
+}
+
+with DAG(
```
Comment (hancush) [inline dags/full_scrape.py:34]: Note that I could revise the config object structure and scrape factory to allow for DAGs with multiple tasks in order to cover the full scrape and hourly ETL, but I actually like the simplicity they have now, so I'd prefer to explicitly define the DAGs with multiple tasks, for now.

File: dags/constants.py [hunk 3]
```diff
@@ -30,30 +30,14 @@
...
-    'windowed_event_scraping': 'Scrape events with a window of 0.05 at 0, 15, 30, and 45 minutes past the hour. Between 9 p.m. UTC Friday and 6 a.m. UTC Saturday, scrape events with a window of 1 at 35 and 50 minutes past the hour. Windowed scrapes capture events with timestamps within a given window or in the future. This generally takes somewhere between a few seconds and a few minutes, depending on the volume of updates.',
-    'fast_full_scraping': 'Scrape all events quickly on the hour and all bills quickly at 5 past the hour between 9 p.m. UTC Friday and 6 a.m. UTC Saturday. Fast scrapes scrape all bills or events quickly – that is, with requests issues as quickly as the server will respond to them. This generally takes less than 30 minutes.',
+    'full_scrape': 'Scrape all people and committees, bills, and events "politely" – that is, with requests throttled to 60 per minute, or 1 per second. This generally takes 6-7 hours.',
```
Comment (fgregg) [inline dags/constants.py:33]: seems like we could nicely defactor these into config.py and full_scrape.py

**Review by fgregg — COMMENTED** [review]
looking really good. one requested change

File: dags/constants.py (hunk 3 above)
Comment (hancush) [reply dags/constants.py:33]: Great point. Doing...

Comment (hancush) [reply dags/constants.py]: Done in https://github.com/datamade/la-metro-dashboard/pull/62/commits/cb757f74891bd0fdc64285df045bd2b727dbb4aa!

**Review by fgregg — APPROVED** [review]

**Pull request merged**